Automattic/woocommerce-payments
# Simplify Deposits UI – Refactor AccountBalancesTabPanel loading state to ensure tooltips remain open

Fixes #6020

If a tooltip in the new Account Balances card is visible when the recent deposits card completes loading, the tooltip will unexpectedly close.

This occurs is because the the Account Overview state is updated after the recent deposits fetch is completed, causing the `AccountBalancesTabPanel` component to fully re-render (including children, i.e. tooltips). 

The TabPanel was intentionally being forced to re-render ([ref](https://github.com/Automattic/woocommerce-payments/pull/5822/files#diff-3244e41cc9e47bf7ef21b95f428c1512cf251d5bae8761ea13c176878799da2aR92-R93)), since the TabPanel needs to re-render when the Account Overview has finished loading in order to select the currently selected currency. This is because TabPanel is not a controlled React component and only accepts an `initialTabName` prop.

This PR returns a loading `AccountBalancesTabPanel` component early, which allows the TabPanel to select the correct currency once loading has completed while avoiding a full component re-render on every subsequent hook state change.

##### Before:

https://user-images.githubusercontent.com/3147296/231634641-ebc1b0ee-7428-49a3-b106-40b6a4e4821d.mov

##### After:

https://user-images.githubusercontent.com/3147296/231634614-c70a5667-c149-4e0c-a7ef-3afc563b70fc.mov


#### Changes proposed in this Pull Request









#### Testing instructions






1. Enable the `_wcpay_feature_simplify_deposits_ui` feature flag.
To enable, run `update_option( '_wcpay_feature_simplify_deposits_ui', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/).
1. View the Payments → Overview screen.
2. While the recent deposits table is loading, click on a tooltip for Available or Pending balance (see screen recording above).
3. On `develop`, this tooltip will disappear once the recent deposits table has completed loading.
3. On this branch, ensure that the tooltip does not disappear once the recent deposits table has completed loading.
1. If you have more than 1 currency enabled, select a tab → refresh the page → ensure that the same currency tab is selected on loading completion.

-------------------


**Post merge**
